src/Pos/Client/Txp/Balances.hs

```diff
@@ -42,62 +39,52 @@ class Monad m => MonadBalances m where
     -- TODO: add a function to get amount of stake (it's different from
     -- balance because of distributions)
 
-    default getOwnUtxos :: (MonadTrans t, MonadBalances m', t m' ~ m) => [Address] -> m Utxo
-    getOwnUtxos = lift . getOwnUtxos
-
-    default getBalance :: (MonadTrans t, MonadBalances m', t m' ~ m) => Address -> m Coin
-    getBalance = lift . getBalance
-
 instance {-# OVERLAPPABLE #-}
     (MonadBalances m, MonadTrans t, Monad (t m)) =>
         MonadBalances (t m)
+  where
+    getOwnUtxos = lift . getOwnUtxos
+    getBalance = lift . getBalance
+
 
 getBalanceFromUtxo :: MonadBalances m => Address -> m Coin
 getBalanceFromUtxo addr =
     unsafeIntegerToCoin . sumCoins .
     map (txOutValue . toaOut) . toList <$> getOwnUtxo addr
 
-data BalancesRedirectTag
+type BalancesMonad ext m =
+    (MonadRealDB m, MonadDBRead m, MonadGState m, MonadMask m, WithLogger m, MonadTxpMem ext m)
 
-type BalancesRedirect =
-    Ether.TaggedTrans BalancesRedirectTag IdentityT
+getOwnUtxosWebWallet :: BalancesMonad ext m => [Address] -> m Utxo
+getOwnUtxosWebWallet addr = do
```

Comment:
1. I don't like `BalancesMonad` name, but I don't have a good alternative. Maybe `MonadBalancesBase`?
2. I don't like `WebWallet` in names. This module doesn't know anything about web wallet. And this function are not specific to web wallet. They can be used by stuff like smart-generator or whatever crazy utility.